src/titanium-data-table.ts

```diff
@@ -0,0 +1,347 @@
+import '../lib/titanium-svg-button';
+import '../lib/titanium-loading-indicator';
+
+import {css, customElement, html, LitElement, property, queryAll} from 'lit-element';
+import {TitaniumDataTableItem} from './titanium-data-table-item';
+
+@customElement('titanium-data-table')
+export class TitaniumDataTable extends LitElement {
+  @property() title: string;
+  @property() take: number;
+  @property() page: number = 0;
+  @property() count: number;
+  @property() items: Array<any>;
+  @property({type: Boolean, attribute: 'single-select', reflect: true})
+  singleSelect: boolean;
+  @property() selected: Array<any> = [];
+  @queryAll('table-container') tableContanier: NodeListOf<any>;
+
+  connectedCallback() {
+    super.connectedCallback();
+    this.setTake(this._determineTake());
+    this.setPage(0);
+    this.addEventListener(
+        'titanium-data-table-item-selected-changed',
+        this._handleItemSelectionChange.bind(this));
+  }
+
+  clearSelection() {
+    this.deselectAll();
+    // Ensure the collection is empty, deselect can cause a race condition
+    // between deselecting and UI drawing new items.
+    this.selected = [];
+    this._notifiySelectedChanged();
+  }
+
+  updated(changedProps) {
+    if (changedProps.has('items') && changedProps.get('items') !== this.items) {
+      // Clear selection when items array changes.
+      this.clearSelection();
+    }
+  }
+
+  private _handleItemSelectionChange(
+      e: CustomEvent<{item: any, isSelected: boolean}>) {
+    if (e.detail.isSelected) {
```

Comment:
Since this event has bubbles, should we stop it here?

Reply:
Yep good idea.